src/site/lab3.rst

```diff
@@ -11,102 +11,53 @@ Lab #3
 
 
-Implementing a Stack
---------------------
+Have a Working Implementation
+-----------------------------
 
-1. Create a project and make a class where you will put your main method
-    * Perhaps call it ``Lab3``
-
-2. Download the :download:`Stack interface.<../main/java/Stack.java>` and add it to your project
-
-3. Create a new class for our ``ArrayStack`` implementation
-
-4. :doc:`Complete your ArrayStack based on the course notes </topic6>`
-
-.. warning::
-
-    I realize you can simply download the :download:`ArrayStack <../main/java/ArrayStack.java>` code and copy/paste, but
-    this defeats the purpose. I want you to slowly and deliberately implement each function and take the time to
-    understand the details.
-
-5. Test your ``ArrayStack`` with the following
-
-    .. code-block:: java
-        :linenos:
-
-            public static void main(String[] args) {
-                Stack<Integer> myStack = new ArrayStack<>();
-
-                System.out.println(myStack);
-
-                for (int i = 0; i < 10; ++i) {
-                    myStack.push(i);
-                }
-
-                System.out.println(myStack);
-
-                while (!myStack.isEmpty()) {
-                    System.out.println(myStack.pop());
-                }
-                System.out.println(myStack);
-            }
-
-
-6. Now create an instance of a stack for characters and rewrite the test code above to test its functionality
+1. If you have not already, finish `:doc: Lab2 </lab2>`
 
 
 Writing Tests
 -------------
 
-1. Write tests for your ``ArrayStack`` implementation
-    * If you do not know how, :doc:`read the aside on testing </topic6-testing>`
-
-.. warning::
-
-    Again, don't simply copy and paste everything from the :download:`ArrayStackTest<../test/java/ArrayStackTest.java>`
-    I provided. Feel free to copy methods one at a time, but he point is to add the tests deliberately to understand
-    what they're doing and what they're testing.
-
-2. Run the tests and make sure they all pass
-    * If they do not, figure out why
-        * Are your tests wrong?
-        * Is your implementation of ``ArrayStack`` wrong?
-
-
-Reversing a String
-------------------
+**Tips**
+* Make your test method names as descriptive as possible while still being brief
+* Follow some convention for your method names, something like ``methodConditionExpected()``
+    * ``sizeOfEmptyCourseListReturnsZero()``
+    * ``addingBeyondCapacityCallsExpandCapacityToMakeRoomForAdd``
 
-1. Write a method that takes a string and returns a new string that is the reverse of the original
-    * This function **must** use a stack for the reversing of the string
-        * **Hint:** Remember what LIFO order is
-    * I have provided you with a skeleton of the method and some simple testing code below
 
+1. Create your testing class for ``CourseList`` called ``CourseListTest``
+    * We will not test ``Course`` since it is so simple
```

Comment:
Simple classes still need tests, contrary to popular belief. It's fine to not write tests, but I'd rather see this worded as not being required _for this lab_.